lib/service/desktop.go

```diff
@@ -260,14 +260,14 @@ func (process *TeleportProcess) initWindowsDesktopServiceRegistered(log *logrus.
 
 	// Cleanup, when process is exiting.
 	process.OnExit("windows_desktop.shutdown", func(payload interface{}) {
+		warnOnErr(listener.Close(), log)
```

Comment:
If it's important that this goes first, add a comment explaining why for future readers/people who may be refactoring this.

Reply:
Reduced the scope of this PR to just the proxy, as it's very much a special case and everything else ultimately works fine right now (other than race conditions related to shutting down before or during init, which the current PR wouldn't address anyway). I think there's a better solution for "listener cleanliness" to be managed at the `TeleportProcess` level anyway (by forcibly closing every exported/exportable listener at shutdown).